File: CasaBackend/Casa.Core/Models/Automation.cs
namespace CasaBackend.Casa.Core.Models
{
    public class Automation
    {
        public int Id { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public bool State { get; set; }
        public TimeSpan InitTime { get; set; }
        public TimeSpan EndTime { get; set; }
        public List<AutomationDevice> Devices { get; set; } = new List<AutomationDevice>();
    }
}

File: MQTT/types/automation.go
package types

type Automation struct {
	ID          int      `json:"id"`
	State       bool     `json:"state"`
	StateText   string   `json:"stateText"`
	Name        string   `json:"name"`
	Description string   `json:"description"`
	InitTime    string   `json:"initTime"`
	EndTime     string   `json:"endTime"`
	Devices     []Device `json:"devices"`
}
